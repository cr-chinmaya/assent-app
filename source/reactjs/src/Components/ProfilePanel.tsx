import { Panel, PersonaSize, Separator } from '@fluentui/react';
import * as React from 'react';
import { useDynamicReducer } from '@micro-frontend-react/employee-experience/lib/useDynamicReducer';
import { Context, withContext } from '@micro-frontend-react/employee-experience/lib/Context';
import { IEmployeeExperienceContext } from '@micro-frontend-react/employee-experience/lib/IEmployeeExperienceContext';
import { getIsProfilePanelOpen, getProfile } from './Shared/SharedComponents.selectors';
import { ActionButton } from '@fluentui/react/lib/Button';
import { toggleProfilePanel } from './Shared/SharedComponents.actions';
import { Persona } from './Shared/Components/Persona';

export function ProfilePanel(): React.ReactElement {
    const { useSelector, dispatch, authClient } = React.useContext(
        Context as React.Context<IEmployeeExperienceContext>
    );
    const isProfilePanelOpen = useSelector(getIsProfilePanelOpen);
    const profile = useSelector(getProfile);

    const handleProfileDismiss = (): void => {
        dispatch(toggleProfilePanel(false));
    };

    const handleLogout = (): void => {
        authClient.logOut();
    };

    return (
        <Panel
            isLightDismiss
            isOpen={isProfilePanelOpen}
            onDismiss={handleProfileDismiss}
            closeButtonAriaLabel="Close profile panel"
            headerText="Profile"
        >
            <Persona
                emailAlias={profile?.userPrincipalName}
                size={PersonaSize.size56}
                text={profile?.displayName}
                secondaryText={profile?.userPrincipalName}
                styles={{ root: { paddingTop: '20px' } }}
            />
            <Separator />
            <ActionButton styles={{ label: { color: '#0078d4' }, root: { marginLeft: '-5px' } }} onClick={handleLogout}>
                Sign out
            </ActionButton>
        </Panel>
    );
}
