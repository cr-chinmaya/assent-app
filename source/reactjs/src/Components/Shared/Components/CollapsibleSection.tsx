import * as React from 'react';
import { Stack } from '@fluentui/react/lib/Stack';
import { IconButton, IStyle, Separator, Text } from '@fluentui/react';

export function CollapsibleSection(props: { titleText: string; children: any; style?: IStyle }): React.ReactElement {
    const { titleText, children, style } = props;
    const [isExpanded, setIsExpanded] = React.useState(false);

    const toggleSection = () => {
        setIsExpanded(!isExpanded);
    };
    return (
        <Stack>
            <Stack.Item>
                <Stack horizontal styles={{ root: style }}>
                    <Stack.Item>
                        <IconButton
                            iconProps={{ iconName: isExpanded ? 'ChevronUp' : 'ChevronDown' }}
                            title={isExpanded ? 'Collapse section' : 'Expand section'}
                            onClick={toggleSection}
                        />
                    </Stack.Item>
                    <Stack.Item styles={{ root: { paddingTop: '5px', paddingLeft: '5px' } }}>
                        <Text styles={{ root: { fontWeight: 600 } }}> {titleText} </Text>
                    </Stack.Item>
                </Stack>
            </Stack.Item>
            {isExpanded && <Stack.Item>{children}</Stack.Item>}
        </Stack>
    );
}
