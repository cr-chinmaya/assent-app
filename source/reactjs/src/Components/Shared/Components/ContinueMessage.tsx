import * as React from 'react';
import { Label, PrimaryButton, Stack } from '@fluentui/react';
import { withContext } from '@micro-frontend-react/employee-experience/lib/Context';
import { Context } from '@micro-frontend-react/employee-experience/lib/Context';
import { getUserAlias } from '../SharedComponents.persistent-selectors';
import {
    refreshBulkState,
    requestMySummary,
    updatePanelState,
    updateIsProcessingBulkApproval,
} from '../SharedComponents.actions';
import { useHistory, useLocation } from 'react-router-dom';
import { IEmployeeExperienceContext } from '@micro-frontend-react/employee-experience/lib/IEmployeeExperienceContext';

export function ContinueMessage(props: { isBulkAction: boolean; customLabel?: string }): React.ReactElement {
    const { useSelector, dispatch } = React.useContext(Context as React.Context<IEmployeeExperienceContext>);
    const isBulkAction = props.isBulkAction;
    const customLabel = props.customLabel;
    const userAlias = useSelector(getUserAlias);
    const history = useHistory();
    const location = useLocation();

    function handleContinueClick(): void {
        if (history && location) {
            if (location.pathname.length > 1) {
                history.push('/');
            }
        }
        if (isBulkAction) {
            dispatch(updateIsProcessingBulkApproval(false));
            dispatch(refreshBulkState());
            dispatch(requestMySummary(userAlias));
        } else {
            dispatch(updatePanelState(false));
            dispatch(requestMySummary(userAlias));
        }
    }

    return (
        <div
            style={{
                height: '100%',
                width: '100%',
                display: 'flex',
                justifyContent: 'center',
                alignItems: 'center',
                outline: 'none',
            }}
            ref={(input) => input && input.focus()}
            tabIndex={0}
        >
            <Stack tokens={{ childrenGap: '10' }}>
                <Stack.Item>
                    <Label>
                        {customLabel ?? 'Please wait for confirmation or click continue to work on other requests.'}
                    </Label>
                </Stack.Item>
                <Stack.Item align={isBulkAction ? 'end' : 'center'}>
                    <PrimaryButton ariaLabel="Continue" text="Continue" onClick={handleContinueClick} />
                </Stack.Item>
            </Stack>
        </div>
    );
}
