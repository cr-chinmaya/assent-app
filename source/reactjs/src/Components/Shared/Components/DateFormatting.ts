export function mapDate(date: string){
    const monthNames = [
        "Jan ", "Feb ", "Mar ",
        "Apr ", "May ", "June ", "July ",
        "Aug ", "Sept ", "Oct ",
        "Nov ", "Dec "
      ];
    var [year, monthIndex, day] = date.substring(0, 10).split("-");
    var month = monthNames[parseInt(monthIndex, 10)-1];
    return month + day.toString() + ", " + year.toString();
}