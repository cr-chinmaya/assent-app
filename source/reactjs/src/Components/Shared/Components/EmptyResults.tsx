import * as React from 'react';
import * as Styled from '../SharedLayout';
import { withContext } from '@micro-frontend-react/employee-experience/lib/Context';
import { Stack } from '@fluentui/react';

function EmptyResults(props: { message?: string }): React.ReactElement {
    const displayMessage = props.message? props.message : "No Results Found";

    const [emptyResultsRef, setEmptyResultsRef] = React.useState(null);

    React.useEffect(() => {
        if (emptyResultsRef) {
            emptyResultsRef.focus();
        }
    }, [emptyResultsRef]);


    return (
        <div style={{height: '100%', width: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center'}}>
            <Stack horizontalAlign='center'>
                <Stack.Item grow>
                    <Styled.EmptySearchIcon title="Empty Search Icon" aria-label="Empty Search Icon" role="text" />
                </Stack.Item>
                <Stack.Item>
                <div
                    ref={(input: any): void => {
                            setEmptyResultsRef(input);
                        }}
                    role="status"
                    aria-label={displayMessage}
                    title={displayMessage}
                    tabIndex={0}
                    style={{ outline: 'none' }}
                >
                    <Styled.MessageBarTitle> {displayMessage} </Styled.MessageBarTitle>
                    </div>
                </Stack.Item>
            </Stack>
        </div>
    );
}

const connected = withContext(EmptyResults);
export { connected as EmptyResults };