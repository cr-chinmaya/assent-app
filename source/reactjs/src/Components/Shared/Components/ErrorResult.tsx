import * as React from 'react';
import * as Styled from '../SharedLayout';
import { Stack } from '@fluentui/react';

function ErrorResult(props: { message?: string }): React.ReactElement {
    return (
        <div style={{ height: '100%', width: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
            <Stack horizontalAlign="center">
                <Stack.Item grow>
                    <Styled.ErrorResultIcon title="Error result icon" aria-label="Error result icon" />
                </Stack.Item>
                <Stack.Item>
                    <Styled.MessageBarTitle>
                        {props.message ? props.message : 'An error occured'}
                    </Styled.MessageBarTitle>
                </Stack.Item>
            </Stack>
        </div>
    );
}

export default ErrorResult;
