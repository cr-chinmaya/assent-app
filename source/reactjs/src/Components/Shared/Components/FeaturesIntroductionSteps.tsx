import { IFeaturesIntroductionStep } from "../SharedComponents.types";

export const teachingSteps: IFeaturesIntroductionStep[] = [
    {
        // This step happens when we decline step 1
        step: 0,
        headline: "You can always take the tour later by clicking on help!",
        target: "#help",
        successButtonLabel: "Great!",
        declineButtonLabel: null,
        successNextStep: -1,
        declineNextStep: -1
    },
    {
        step: 1,
        headline: "Welcome to MSApprovals! Want to learn our new features?",
        target: "#MyView_id0",
        successButtonLabel: "Take Tour",
        declineButtonLabel: "Maybe Later",
        successNextStep: 2,
        declineNextStep: 0
    },
    {
        step: 2,
        headline: "Click here to group items!",
        target: "#groupBy",
        successButtonLabel: "Next",
        declineButtonLabel: null,
        successNextStep: 3,
        declineNextStep: -1
    },
    {
        step: 3,
        headline: "Filter your items here!",
        target: "#filter",
        successButtonLabel: "Next",
        declineButtonLabel: null,
        successNextStep: 4,
        declineNextStep: -1
    },
    {
        step: 4,
        headline: "Click here to view your history!",
        target: "#history",
        successButtonLabel: "Next",
        declineButtonLabel: null,
        successNextStep: 5,
        declineNextStep: -1
    },
    {
        step: 5,
        headline: "Thanks for taking the tour!",
        target: "#MyView_id0",
        successButtonLabel: "Done",
        declineButtonLabel: null,
        successNextStep: -1,
        declineNextStep: -1
    }
]

export const initialTeachingStep: number = 1;