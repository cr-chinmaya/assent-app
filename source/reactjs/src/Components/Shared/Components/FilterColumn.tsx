/* eslint-disable react/jsx-key */
import * as React from 'react';
import { Checkbox, IComboBox, IComboBoxOption, Label, Stack, VirtualizedComboBox } from '@fluentui/react';
import BasicButton from '../../../Controls/BasicButton';

export interface IFilterOption {
    label: string;
    checked: boolean;
}

interface IFilterColumnProps {
    columnCategory: { key: string; label: string };
    columnOptions: IFilterOption[];
    selectedKeys: string[];
    onChange: (
        ev: React.FormEvent<HTMLElement>,
        checked: boolean,
        columnCategory: { key: string; label: string },
        optionLabel: string
    ) => void;
    onClear: (columnCategory: { key: string; label: string }) => void;
    showDropdown: boolean;
}

const stackTokens = { childrenGap: 10 };

const isCollapsible = false;
const showClear = true;

function renderComboboxDropdown(
    columnOptions: IFilterOption[],
    selectedKeys: string[],
    onChange: any,
    columnCategory: any
): JSX.Element {
    const comboOptions: IComboBoxOption[] = columnOptions.map((item, index) => ({
        key: item.label,
        text: item.label.toString(),
        selected: item.checked,
        styles: { optionText: { whiteSpace: 'normal' } },
    }));
    return (
        <VirtualizedComboBox
            label={columnCategory.label}
            aria-label={columnCategory.label}
            placeholder="Select values to filter"
            options={comboOptions}
            dropdownMaxWidth={200}
            useComboBoxAsMenuWidth
            multiSelect
            autoComplete="off"
            selectedKey={selectedKeys}
            onChange={(
                event: React.FormEvent<IComboBox>,
                option?: IComboBoxOption,
                index?: number,
                value?: string
            ): void => {
                onChange(event, option.selected, columnCategory, option.key);
            }}
        />
    );
}

export function FilterColumn(props: IFilterColumnProps): React.ReactElement {
    const { columnOptions, columnCategory, onChange, onClear, showDropdown, selectedKeys } = props;
    return (
        <Stack tokens={stackTokens}>
            {!isCollapsible && !showDropdown && (
                <Stack.Item styles={{ root: { paddingLeft: '8%' } }}>
                    <Label id={'comboxbox-' + columnCategory.label?.replace(' ', '')}>{columnCategory.label}</Label>
                </Stack.Item>
            )}
            <Stack.Item>
                <div style={{ paddingLeft: '8%' }}>
                    <Stack tokens={{ childrenGap: 7 }}>
                        {!showDropdown &&
                            columnOptions &&
                            columnOptions.map((item, index) => (
                                <Stack.Item>
                                    <Checkbox
                                        label={item.label.toString()}
                                        checked={item.checked}
                                        onChange={(ev: React.FormEvent<HTMLElement>, checked: boolean) => {
                                            onChange(ev, checked, columnCategory, item.label);
                                        }}
                                    />
                                </Stack.Item>
                            ))}
                        {showDropdown && renderComboboxDropdown(columnOptions, selectedKeys, onChange, columnCategory)}
                    </Stack>
                </div>
            </Stack.Item>
            {showClear && (
                <Stack.Item>
                    <BasicButton
                        primary={false}
                        text="Clear"
                        title="Clear"
                        onClick={() => {
                            onClear(columnCategory);
                        }}
                        styles={{
                            root: { border: 'none', padding: 0 },
                            label: {
                                color: '#0064C1',
                                selectors: {
                                    '@media (min-device-width: 1023px) and (min-width: 320px) and (max-width: 639px)': {
                                        fontSize: '12px',
                                    },
                                },
                            },
                            rootHovered: { backgroundColor: '#fafafa', border: 'none' },
                            rootFocused: { border: '1px solid rgb(177, 177, 177)' },
                        }}
                    />
                </Stack.Item>
            )}
        </Stack>
    );
}
