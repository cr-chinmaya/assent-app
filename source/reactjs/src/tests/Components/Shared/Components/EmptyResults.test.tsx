import * as React from 'react';
import { render } from '@testing-library/react';
import {EmptyResults}  from '../../../../Components/Shared/Components/EmptyResults';

describe("EmptyResults", () => {
    it("should EmptyResults with message renders correctly", () => {
      const wrapper = render(<EmptyResults message={'testmessage'} />)
      expect(wrapper).toMatchSnapshot();
    });

    it("should EmptyResults without message renders correctly", () => {
      const wrapper = render(<EmptyResults />)
      expect(wrapper).toMatchSnapshot();
    });
});