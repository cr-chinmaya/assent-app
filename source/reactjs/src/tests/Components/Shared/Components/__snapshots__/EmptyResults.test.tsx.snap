// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyResults should EmptyResults with message renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="height: 100%; width: 100%; display: flex; justify-content: center; align-items: center;"
      >
        <div
          class="ms-Stack css-40"
        >
          <div
            class="ms-StackItem css-41"
          >
            <div
              aria-label="Empty Search Icon"
              class="sc-iBPRYJ fBnPKg"
              title="Empty Search Icon"
            />
          </div>
          <div
            class="ms-StackItem css-42"
          >
            <p
              class="sc-pFZIQ heQLDe"
            >
               
              testmessage
               
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="height: 100%; width: 100%; display: flex; justify-content: center; align-items: center;"
    >
      <div
        class="ms-Stack css-40"
      >
        <div
          class="ms-StackItem css-41"
        >
          <div
            aria-label="Empty Search Icon"
            class="sc-iBPRYJ fBnPKg"
            title="Empty Search Icon"
          />
        </div>
        <div
          class="ms-StackItem css-42"
        >
          <p
            class="sc-pFZIQ heQLDe"
          >
             
            testmessage
             
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`EmptyResults should EmptyResults without message renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="height: 100%; width: 100%; display: flex; justify-content: center; align-items: center;"
      >
        <div
          class="ms-Stack css-40"
        >
          <div
            class="ms-StackItem css-41"
          >
            <div
              aria-label="Empty Search Icon"
              class="sc-iBPRYJ fBnPKg"
              title="Empty Search Icon"
            />
          </div>
          <div
            class="ms-StackItem css-42"
          >
            <p
              class="sc-pFZIQ heQLDe"
            >
               
              No Results Found
               
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="height: 100%; width: 100%; display: flex; justify-content: center; align-items: center;"
    >
      <div
        class="ms-Stack css-40"
      >
        <div
          class="ms-StackItem css-41"
        >
          <div
            aria-label="Empty Search Icon"
            class="sc-iBPRYJ fBnPKg"
            title="Empty Search Icon"
          />
        </div>
        <div
          class="ms-StackItem css-42"
        >
          <p
            class="sc-pFZIQ heQLDe"
          >
             
            No Results Found
             
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
